interfaces/cython/cantera/ctml2yaml.py

```diff
@@ -0,0 +1,2682 @@
+                    else:
+                        if spec.attribs["electrolyte-species-type"] != etype:
+                            warnings.warn(
+                                "The electrolyteSpeciesType node was specified at the "
+                                "phase and species level for species '{}'. The value "
+                                "specified in the species node will be "
+                                "used".format(name)
+                            )
+
+        return activity_data
+
+    @classmethod
+    def to_yaml(cls, representer, data):
+        """Serialize the class instance to YAML format suitable for ruamel.yaml.
+
+        :param representer:
+            An instance of a ruamel.yaml representer type.
+        :param data:
+            An instance of this class that will be serialized.
+
+        The class instance should have an instance attribute called ``attribs`` which
+        is a dictionary representing the information about the instance. The dictionary
+        is serialized using the ``represent_dict`` method of the ``representer``.
+        """
+        return representer.represent_dict(data.attribs)
+
+
+class SpeciesThermo:
+    def __init__(self, thermo: etree.Element) -> None:
+        """Represent the polynomial-type thermodynamic data for a `Species`.
+
+        :param thermo:
+            A ``species/thermo`` XML node. Must have one or more child nodes with tag
+            ``NASA``, ``NASA9``, ``const_cp``, ``Shomate``, or ``Mu0``.
+
+        This class will process the `Species`-level thermodynamic information for the
+        polynomial thermo types. The pressure-dependent standard state types are
+        processed directly in the `Species` instance.
+        """
+        thermo_type = thermo[0].tag
+        if thermo_type not in ["NASA", "NASA9", "const_cp", "Shomate", "Mu0"]:
+            raise TypeError("Unknown thermo model type: '{}'".format(thermo[0].tag))
+        func = getattr(self, thermo_type)
+        self.attribs = func(thermo)
+
+    def process_polynomial(
+        self, thermo: etree.Element, poly_type: str
+    ) -> Tuple[List[List[float]], List[float]]:
+        """Process the `Species` thermodynamic polynomial for several types.
+
+        :param thermo:
+            A ``species/thermo`` XML node. Must have one or more child nodes with tag
+            ``NASA``, ``NASA9``, or ``Shomate``.
+        :param poly_type:
+            A string determining the type of polynomial. One of ``NASA``, ``NASA9``,
+            or ``Shomate``.
+
+        This method converts the polynomial data for the ``NASA``, ``NASA9``, and
+        ``Shomate`` thermodynamic types into the appropriate YAML structure.
+        """
+        temperature_ranges = set()
+        model_nodes = thermo.findall(poly_type)
+        unsorted_data = {}
+        for node in model_nodes:
+            Tmin = float(node.get("Tmin", 0))
+            Tmax = float(node.get("Tmax
```

Comment:
I checked with the file that @ischoegl sent me that prompted this change and that seems to be correct, only the first one is loaded. What do you think we should do in this case during conversion? Also only take the first one, or just emit a warning? The current behavior seems like the best compromise to me; silently ignoring the second node seems likely to cause confusion, and if we emit a warning anyways, we might as well just assume that the sections should be combined since getting people to improve/correct their XML documents isn't really the point (as you correctly noted earlier).

Reply:
I would say warn and skip the extra sections. Otherwise, you're producing a YAML file that will result in loading a different set of reactions than would have ever been loaded from the XML file.